src/modalities/models/model.py

```diff
@@ -1,14 +1,29 @@
+import math
 from abc import abstractmethod
-from typing import Dict, List, Optional
+from enum import Enum
+from functools import partial
+from typing import Annotated, Dict, List, Optional
 
 import torch
 import torch.nn as nn
+from pydantic import BaseModel, Field
 
 from modalities.batch import DatasetBatch, InferenceResultBatch
 
 WeightDecayGroups = Dict[str, List[str]]
 
 
+class ActivationType(str, Enum):
+    GELU = "gelu"
+    FUSED_SWIGLU = "fused_swiglu"
+
+
+class WeightInitializationConfig(BaseModel):
+    mean: Annotated[float, Field(strict=True, ge=0.0)]
```

Comment:
This will afaik in practice virtually always be set to `0.0`, but yes, I guess in principle it can also be negative, so feel free to change it as part of #168